### PR TITLE
Adding AOAI policies to the GenAI Workload

### PR DESCRIPTION
**Pull request by prasann**
Adding 2 scenarios based on the new set of AOAI policies, along with the infra setup and docs.

-  Moved the existing rate limiting policy as a workaround and kept the new ones as the default
- Added metric tracking via appinsights and made it as a default.

### CODE REVIEW

**Review by aarthiem — APPROVED** [review]
Thanks Prasanna!

File: scenarios/workload-genai/bicep/apim-policies/apiManagement.bicep
Comment (seenu433) [inline scenarios/workload-genai/bicep/apim-policies/apiManagement.bicep:1]: Need to add the diagnostics at the api level

resource apiMonitoring 'Microsoft.ApiManagement/service/apis/diagnostics@2020-06-01-preview' = {
  name: 'applicationinsights'
  parent: azureOpenAIApi
  properties: {
      alwaysLog: 'allErrors'
      loggerId: appInsightsLoggerId
      metrics: true
      logClientIp: true
      httpCorrelationProtocol: 'W3C'
      verbosity: 'information'
      operationNameFormat: 'Url'
  }
}

File: scenarios/workload-genai/policies/fragments/rate-limiting/adaptive-rate-limiting.xml
Comment (seenu433) [inline scenarios/workload-genai/policies/fragments/rate-limiting/adaptive-rate-limiting.xml:1]: Not sure of the behavior when multiple policy blocks for rate limiting are applied. also the remainingtokens variable is expected to hold the tokens left for that subscriber alone.

File: scenarios/scripts/deploy-apim-baseline.sh
Comment (seenu433) [inline scenarios/scripts/deploy-apim-baseline.sh:1]: This location constraint will impact all deployments scenarios. Probably we need to add a description that it is temporary and used for the genai dependencies

File: scenarios/.env
Comment (seenu433) [inline scenarios/.env:1]: we need to ignore changes to this file. the RANDOM_IDENTIFIER will be autogenerated for each deployment session

**Review by seenu433 — CHANGES_REQUESTED** [review]

File: scenarios/workload-genai/bicep/apim-policies/apiManagement.bicep
Comment (prasann) [reply scenarios/workload-genai/bicep/apim-policies/apiManagement.bicep:1]: I was thinking of this approach, in this case, we need to pass on the appInsights  value here. but looks like this wasn't needed. the service level diagnostic setting was inherited and worked fine.

**Review by seenu433 — APPROVED** [review]

**Pull request merged**